src/network/core/packet.cpp

```diff
@@ -57,7 +58,7 @@ Packet::Packet(PacketType type, size_t limit) : pos(0), limit(limit), cs(nullptr
  */
 void Packet::PrepareToSend()
 {
-	assert(this->cs == nullptr);
+	assert(this->buffer[0] == 0 && this->buffer[1] == 0);
```

Comment:
This change feels unexpected? Should it be part of this commit? And what is the reason behind it?

I do agree buffer[0] and [1] should be zero. But why no longer check if cs isn't nullptr? The Packet could still be made with a nullptr, so it is good check, not?

Reply:
On the other hand, `cs` is not used in this function, so that is a bit odd. Guess mostly it felt random in this commit that made me wonder. But I guess it also doesn't hurt. So if this was intended, I am fine with this.

Reply:
The `assert(this->cs == nullptr)` was to prevent this function from being called on a received packet, because `cs` was `nullptr` when sending the packet.
Since `cs` can now also be `nullptr`, I needed another method. Upon creation of the packet, two zero bytes are reserved to later save the size of the packet. That's those two bytes. When you receive a packet,  they will not be both zero as you interpret the size in the code reading the bytes from the socket.

So the new assertion is slightly stronger, as it will fail upon calling `PrepareToSend` when the packet was received from the network, but it will also prevent `PrepareToSend` from being called twice.

Reply:
So in that logic, line 247 was to ensure it was done on a received packet, and not on a write packet. That was already unclear (lack of comments anyone? :D ), but maybe it would be good to also address that in this PR, by the same logic as this change?

I can imagine us just removing the line, as it is rather pedantic to have it as this stage. But maybe you have something clever to know if it is actually reading or writing?

Reply:
Removing the line 247 one would be the best way going forward, as I can't do the all zero's trick as then some lovely person is going to send two zeros as size and crash the server/client when assertions are enabled.